ui/app/components/input_output/content_blocks/ToolPayload.tsx

```diff
@@ -0,0 +1,72 @@
+import { CodeEditor, useFormattedJson } from "~/components/ui/code-editor";
+import { Input } from "~/components/ui/input";
+
+interface ToolPayloadProps {
+  id: string;
+  name: string;
+  payload: string;
+  payloadLabel: string;
+  isEditing?: boolean;
+  onChange?: (id: string, name: string, payload: string) => void;
+  enforceJson?: boolean;
+}
+
+// `ToolPayload` renders the content of:
```

Comment:
it was clear after a minute's thought but do you mind adding a comment explaining why we need this component? (shared editing logic with metadata fields)